portal/portal-api/api/src/java/org/sakaiproject/portal/beans/bullhornhandlers/GradeAssignmentBullhornHandler.java

```diff
@@ -70,7 +70,10 @@
                 List<BullhornData> bhEvents = new ArrayList<>();
                 submission.getSubmitters().forEach(to -> {
                     try {
-                        bhEvents.add(new BullhornData(from, to.getSubmitter(), siteId, title, assignmentService.getDeepLink(siteId, assignment.getId(), to.getSubmitter())));
+                        String url = assignmentService.getDeepLink(siteId, assignment.getId(), to.getSubmitter());
+                        if (!url.isEmpty()) {
+                            bhEvents.add(new BullhornData(from, to.getSubmitter(), siteId, title, url));
+                        }
```

Comment:
You need a log statement in here to indicate that the submitter maybe didn't have permissions on the submission. Maybe a TODO about fixing assignments, too.